modules/affile/directory-monitor-factory.c

```diff
@@ -63,6 +70,15 @@ directory_monitor_factory_get_constructor(DirectoryMonitorOptions *options)
     {
       constructor = directory_monitor_poll_new;
     }
+#elif SYSLOG_NG_HAVE_KQUEUE
+  if (options->method == MM_AUTO || options->method == MM_KQUEUE)
+    {
+      constructor = directory_monitor_kqueue_new;
```

Comment:
You should return here, or make sure you don't override it in case of `options->method == MM_AUTO` as it will become poll at line 85

Reply:
sorry never mind I missed the preprocessor else